x/incentive/keeper/keeper.go

```diff
@@ -39,10 +45,267 @@ func NewKeeper(
+
+			// Calculate new uncommitted Eden-Boost tokens for staker and Eden token holders
+			new_uncommitted_eden_boost_tokens := k.CalculateNewUncommittedEdenBoostTokens(ctx, delegatedAmt, commitments, epochIdentifier, stakeIncentive, lpIncentive)
+
+			// Update Commitments with new uncommitted token amounts
+			k.UpdateCommitments(ctx, creator, new_uncommitted_eden_tokens, new_uncommitted_eden_boost_tokens)
+
+			return false
+		},
+	)
+}
+
+// Calculate new Eden token amounts based on the given conditions and user's current uncommitted token balance
+func (k Keeper) CalculateNewUncommittedEdenTokens(ctx sdk.Context, delegatedAmt sdk.Int, commitments ctypes.Commitments, stakeIncentive types.IncentiveInfo, lpIncentive types.IncentiveInfo) sdk.Int {
+	// Get LP commitments - Skip for now
+	edenCommittedByLP := sdk.ZeroInt()
+
+	// Get eden commitments and eden boost commitments
+	edenCommitted := sdk.ZeroInt()
+	edenBoostCommitted := sdk.ZeroInt()
+	for _, c := range commitments.CommittedTokens {
+		// Eden committed
+		if c.Denom == types.Eden {
+			edenCommitted = c.Amount
+		}
+
+		// Eden boost committed
+		if c.Denom == types.EdenB {
+			edenBoostCommitted = c.Amount
+		}
+	}
+
+	// compute eden reward based on above and param factors for each
+	totalEdenCommittedByStake := delegatedAmt.Add(edenCommitted).Add(edenBoostCommitted).Add(edenCommittedByLP)
+	stakeShare := k.CalculateTotalShareOfStaking(totalEdenCommittedByStake)
+
+	// Calculate eden amount per epoch
+	edenAmountPerEpoch := stakeIncentive.Amount.Quo(sdk.NewInt(stakeIncentive.NumEpochs))
+
+	// Calculate newly creating eden amount by its share
+	newEdenAllocated := stakeShare.MulInt(edenAmountPerEpoch)
+
+	return newEdenAllocated.TruncateInt()
+}
+
+// Calculate epoch counts per year to be used in APR calculation
+func (k Keeper) CalculateEpochCountsPerYear(epochIdentifier string) int64 {
+	switch epochIdentifier {
+	case etypes.WeekEpochID:
+		return types.WeeksPerYear
+	case etypes.DayEpochID:
+		return types.DaysPerYear
+	case etypes.HourEpochID:
+		return types.HoursPerYear
+	}
+
+	return 0
+}
+
+// Calculate new Eden-Boost token amounts based on the given conditions and user's current uncommitted token balance
+func (k Keeper) CalculateNewUncommittedEdenBoostTokens(ctx sdk.Context, delegatedAmt sdk.Int, commitments ctypes.Commitments, epochIdentifier string, stakeIncentive types.IncentiveInfo, lpIncentive types.IncentiveInfo) sdk.Int {
+	// Get eden commitments
+	edenCommitted := sdk.ZeroInt()
+	for _, c := range commitments.CommittedTokens {
+		// Eden committed
+		if c.Denom == types.Eden {
+			edenCommitted = c.Amount
+		}
+	}
+
+	// Gompute eden reward based on above and param factors for each
+	totalEden := delegatedAmt.Add(edenCommitted)
+
+	// Calculate 100% APR for eden boost
+	epochNumsPerYear := k.CalculateEpochCountsPerYear(epochIdentifier)
+
+	return totalEden.Quo(sdk.NewInt(epochNumsPerYear))
```

Comment:
This looks correct based on my understanding of the requirements, but will clarify with the product guys.

Reply:
@kenta-elys The calculation here is correct, we just need to add a parameter for the APR percentage and include that in the computation. The case we're talking about here would just have it at 100% but we need to be able to change it in the future.

Reply:
Okay got it.